src/tribler-core/tribler_core/components/implementation/ipv8.py

```diff
@@ -0,0 +1,130 @@
+            endpoint = DispatcherEndpoint([])
+        else:
+            # IPv8 includes IPv6 support by default.
+            # We only load IPv4 to not kill all Tribler overlays (currently, it would instantly crash all users).
+            # If you want to test IPv6 in Tribler you can set ``endpoint = None`` here.
+            endpoint = DispatcherEndpoint(["UDPIPv4"], UDPIPv4={'port': port,
+                                                                'ip': address})
+        ipv8 = IPv8(ipv8_config_builder.finalize(),
+                    enable_statistics=config.ipv8.statistics and not config.gui_test_mode,
+                    endpoint_override=endpoint)
+        await ipv8.start()
+        self.ipv8 = ipv8
+
+        masterkey = await self.use(MasterKeyComponent)
+
+        self.peer = Peer(masterkey.keypair)
+        # self.provide(mediator, ipv8)
+
+        if config.ipv8.statistics and not config.gui_test_mode:
+            # Enable gathering IPv8 statistics
+            for overlay in ipv8.overlays:
+                ipv8.endpoint.enable_community_statistics(overlay.get_prefix(), True)
+
+        if config.ipv8.walk_scaling_enabled and not config.gui_test_mode:
+            from tribler_core.modules.ipv8_health_monitor import IPv8Monitor
+            IPv8Monitor(ipv8,
+                        config.ipv8.walk_interval,
+                        config.ipv8.walk_scaling_upper_limit).start(self.task_manager)
+
+        rest_manager.get_endpoint('statistics').ipv8 = ipv8
+
+        self.bootstrapper = self.peer_discovery_community = self.dht_discovery_community = None
+        self.init_dht_discovery_community()
+        if not self.session.config.gui_test_mode:
+            self.init_bootstrapper()
+            self.init_peer_discovery_community()
+        else:
+            self.dht_discovery_community.routing_tables[UDPv4Address] = RoutingTable('\x00' * 20)
+
+        endpoints_to_init = ['/asyncio', '/attestation', '/dht', '/identity',
+                             '/isolation', '/network', '/noblockdht', '/overlays']
+
+        for path, endpoint in rest_manager.get_endpoint('ipv8').endpoints.items():
+            if path in endpoints_to_init:
+                endpoint.initialize(ipv8)
+
+    def init_bootstrapper(self):
+        args = DISPERSY_BOOTSTRAPPER['init']
+        if bootstrap_override := self.session.config.ipv8.bootstrap_override:
+            address, port = bootstrap_override.split(':')
+            args = {'ip_addresses': [(address, int(port))], 'dns_addresses': []}
+        self.bootstrapper = DispersyBootstrapper(**args)
+
+    def init_peer_discovery_community(self):
+        ipv8 = self.ipv8
+        community = DiscoveryCommunity(self.peer, ipv8.endpoint, ipv8.network, max_peers=100)
+        ipv8.strategies.append((RandomChurn(community), INFINITE))
+        ipv8.strategies.append((PeriodicSimilarity(community), INFINITE))
+        ipv8.strategies.append((RandomWalk(community), INFINITE))
```

Comment:
```suggestion
        ipv8.strategies.append((RandomWalk(community), 20))
```

The current PR code would disproportionally (regarding the connections in other network overlays) introduce you to peers you're not interested in. Of course you want some random connections, but not an "infinite" amount.

Trivia: 20 is the default setting.